SparseGrids/tsgLinearSolvers.cpp

```diff
@@ -167,6 +167,69 @@ void TasmanianTridiagonalSolver::decompose(int n, double d[], double e[], double
     }
 }
 
+void TasmanianFourierTransform::dft(const int rank, const int n[], const std::complex<double> data[], std::complex<double> out[], const int direction){
+    // Fourier transform code by ZBM
+    // DOES NOT NORMALIZE the outgoing Fourier coefficients
+
+    if (direction != 1 && direction != -1) std::cerr << "ERROR: direction must be -1 (forward) or 1 (backward); value is " << direction << std::endl;
```

Comment:
* do not use simple integers as flags in C++, it is OK for C or Fortran, but C++ has more elegant ways, e.g., boolean or enumerate both of which will make the code more readable
* do not change this now, the whole function will be scrapped anyway